pkg/marshal/marshal.go

```diff
@@ -162,3 +163,77 @@ func SmartMarshal[I any](inputfunc interface{}, data I) (output interface{}, err
 	}
 	return nil, ErrNotValidFunc
 }
+
+// UnmarshalMapToStruct unmarshals a map[string]interface{} to a struct.
+func UnmarshalMapToStruct(data map[string]interface{}, outStruct interface{}) error {
```

Comment:
I'm ok with this moving, but we need to remember that this will change with the introduction of Column and Rows

Reply:
Yes, of course, but for now, we should continue to improve the project in meanwhile.